core/trino-main/src/main/java/io/trino/metadata/SystemSecurityMetadata.java

```diff
@@ -240,4 +243,19 @@ default void validateEntityKindAndPrivileges(Session session, String entityKind,
      * to be fully qualified, i.e., if the entity is a table, the name is of size three.
      */
     void setEntityOwner(Session session, EntityKindAndName entityKindAndName, TrinoPrincipal principal);
```

Comment:
Don't we need a generic function like this one? Also I am not a fun that in one place we call it "authorization" and in other "owner". Maybe we should rename `setEntityOwner` to `setEntityAuthorization`?

```
Map<String, TrinoPrincipal> getEntityAuthorizations(Session session, String entityKind);
Optional<TrinoPrincipal> getEntityAuthorization(Session session, EntityKindAndName entityKindAndName);
```
?

Also, if we would go this path then it means we just need `system.metadata.entity_authorization` table.

Reply:
We discussed this offline and we believe making this generic, but we found problems like how to expose an entity in such table in generic way. Like not all entities have table name or schema name.